src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/HistoryTabPanel.kt

```diff
@@ -1,41 +1,62 @@
 package com.atlassian.performance.tools.jiraactions.api.page
 
 import org.openqa.selenium.By
+import org.openqa.selenium.Keys
 import org.openqa.selenium.WebDriver
+import org.openqa.selenium.interactions.Actions
 import org.openqa.selenium.support.ui.ExpectedConditions
 import java.time.Duration
 
 class HistoryTabPanel(
     private val driver: WebDriver
 ) {
+    // the duration is set to such value because viewing issue with many change history entries is a heavy operation,
+    // and we assume the user is patient enough to wait for all entries to show up
+    private val duration: Duration = Duration.ofMinutes(2)
     private val loadMoreEntriesLocator = By.cssSelector("button.show-more-changehistory-tabpanel")
+    private val attemptLimit = 5
 
-    fun loadAllHistoryEntries(): HistoryTabPanel {
+    fun showAllHistoryEntries(): HistoryTabPanel {
         if (driver.isElementPresent(loadMoreEntriesLocator)) {
-            val loadMoreButton = driver.findElement(loadMoreEntriesLocator)
-            loadMoreButton.click()
-            driver.wait(
-                timeout = Duration.ofSeconds(60),
-                condition = ExpectedConditions.stalenessOf(loadMoreButton)
-            )
-        } else {
-            driver.wait(
-                timeout = Duration.ofSeconds(60),
-                condition = ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(".issuePanelWrapper.loading"))
-            )
+            return showMoreEntriesIfNeeded()
         }
+        driver.wait(
+            duration,
+            ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(".issuePanelWrapper.loading"))
+        )
         return this
     }
 
     fun waitForActive(): HistoryTabPanel {
         driver.wait(
-            Duration.ofSeconds(45),
+            duration,
             ExpectedConditions.elementToBeClickable(By.id("changehistory-tabpanel"))
         ).click()
         driver.wait(
-            Duration.ofSeconds(60),
+            duration,
             ExpectedConditions.presenceOfElementLocated(By.cssSelector("#changehistory-tabpanel.active-tab"))
         )
         return this
     }
+
+    private fun showMoreEntriesIfNeeded(): HistoryTabPanel {
+        var attemptCount = 0;
+        do {
+            val loadMoreButton = driver.findElement(loadMoreEntriesLocator)
+            Actions(driver)
+                .keyDown(Keys.SHIFT)
+                .click(loadMoreButton)
+                .keyUp(Keys.SHIFT)
+                .build()
+                .perform()
+            driver.wait(
+                duration,
+                ExpectedConditions.stalenessOf(loadMoreButton)
+            )
+            attemptCount++
+        } while (
+            attemptCount < attemptLimit && driver.isElementPresent(loadMoreEntriesLocator)
```

Comment:
Nit: `for` loop with conditional `break` could be a bit more clean/safe here. Using `while` is always worrying me about deadlocks.

TIL: Apparently Kotlin doesn't have `break`, however `return` with labels can be used to achieve the same capability. This approach is compatible with other alternatives to simple `for` like `repeat()`: https://kotlinlang.org/docs/returns.html#return-to-labels

In this case it would be possible to simply `return this` in the scope of the full method - no labels needed.